### PR TITLE
python3Packages.PyICU: 2.3.1 -> 2.6

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
https://github.com/ovalhub/pyicu/issues/143

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117508` at dda166af run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>gramps</li>
    <li>python38Packages.PyICU</li>
    <li>python38Packages.slob</li>
    <li>python39Packages.PyICU</li>
    <li>python39Packages.slob</li>
    <li>xdxf2slob</li>
  </ul>
</details>

**Review by dotlambda — CHANGES_REQUESTED** [review]
I would rather pass `icu67` as argument.

**Comment by dotlambda** [comment]
Can't we just update pyicu to the latest version?

**Comment by rmcgibbo** [comment]
> Can't we just update pyicu to the latest version?

Yes! That is a much better idea https://github.com/ovalhub/pyicu/issues/143

**Comment by rmcgibbo** [comment]
(Reran nixpkgs-review, since I changed the PR completely. Still looks fine.)

-------

Result of `nixpkgs-review pr 117508` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>gramps</li>
    <li>python38Packages.PyICU</li>
    <li>python38Packages.slob</li>
    <li>python39Packages.PyICU</li>
    <li>python39Packages.slob</li>
    <li>xdxf2slob</li>
  </ul>
</details>

File: pkgs/development/python-modules/pyicu/default.nix [hunk 1]
```diff
@@ -1,33 +1,23 @@
...
-  ];
-
   nativeBuildInputs = [ icu ]; # for icu-config
   buildInputs = [ icu ];
-  checkInputs = [ pytest ];
-  propagatedBuildInputs = [ six ];
+  checkInputs = [ pytestCheckHook six ];
 
   meta = with lib; {
     homepage = "https://pypi.python.org/pypi/PyICU/";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyicu/default.nix:23]:
```suggestion
    homepage = "https://github.com/ovalhub/pyicu/";
```

File: pkgs/development/python-modules/pyicu/default.nix [hunk 2]
```diff
@@ -1,33 +1,23 @@
...
-            + "debian/patches/icu_test.patch";
-      sha256 = "1iavdkyqixm9i753svl17barla93b7jzgkw09dn3hnggamx7zwx9";
-    })
-  ];
-
   nativeBuildInputs = [ icu ]; # for icu-config
   buildInputs = [ icu ];
-  checkInputs = [ pytest ];
-  propagatedBuildInputs = [ six ];
+  checkInputs = [ pytestCheckHook six ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyicu/default.nix:20]: I just checked and six is only used in the tests.

**Review by dotlambda — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]
```
gramps: Ask if output path share/doc (3.5M) could be split with outputs = [ ... "doc" ]
```

**Pull request merged**